shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/adapter/AbstractConnectionAdapter.java

```diff
@@ -239,6 +239,16 @@ public final void setTransactionIsolation(final int level) throws SQLException {
         recordMethodInvocation(Connection.class, "setTransactionIsolation", new Class[]{int.class}, new Object[]{level});
         forceExecuteTemplate.execute(cachedConnections.values(), connection -> connection.setTransactionIsolation(level));
     }
+
+    @Override
+    public final boolean isValid(final int timeout) throws SQLException {
+        for (Connection connection : cachedConnections.values()) {
+            if (!connection.isValid(timeout)) {
+                return false;
```

Comment:
is there some cases need return true, such as m-s?

Reply:
Do you mean when master is up and slave goes down, it should return true?
This seems better, i will try to identify which connections are from the master data source.